src/components/ButtonWithDropdownMenu.js

```diff
@@ -0,0 +1,132 @@
+import React, {useState, useRef, useEffect} from 'react';
+import PropTypes from 'prop-types';
+import {View, useWindowDimensions} from 'react-native';
+import _ from 'underscore';
+import styles from '../styles/styles';
+import Button from './Button';
+import PopoverMenu from './PopoverMenu';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import themeColors from '../styles/themes/default';
+
+const propTypes = {
+    /** Text to display for the menu header */
+    menuHeaderText: PropTypes.string,
+
+    /** Callback to execute when the main button is pressed */
+    onPress: PropTypes.func.isRequired,
+
+    /** Whether we should show a loading state for the main button */
+    isLoading: PropTypes.bool,
+
+    /** Should the confirmation button be disabled? */
+    isDisabled: PropTypes.bool,
+
+    /** Menu options to display */
+    /** e.g. [{text: 'Pay with Expensify', icon: Wallet}, {text: 'PayPal', icon: PayPal}] */
+    options: PropTypes.arrayOf(
+        PropTypes.shape({
+            value: PropTypes.string.isRequired,
+            text: PropTypes.string.isRequired,
+            icon: PropTypes.elementType,
+            iconWidth: PropTypes.number,
+            iconHeight: PropTypes.number,
+            iconDescription: PropTypes.string,
+        }),
+    ).isRequired,
+};
+
+const defaultProps = {
+    isLoading: false,
+    isDisabled: false,
+    menuHeaderText: '',
+};
+
+const ButtonWithDropdownMenu = (props) => {
+    const [selectedItemIndex, setSelectedItemIndex] = useState(0);
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [popoverAnchorPosition, setPopoverAnchorPosition] = useState(null);
+
+    const {width, height} = useWindowDimensions();
```

Comment:
I know I wrote this code, but I think this would improve clarity a bit wherever these values are used:

```suggestion
    const {width: windowWidth, height: windowHeight} = useWindowDimensions();
```

Reply:
Yep, will clean up and comment things later